src/background.js

```diff
@@ -29,50 +29,74 @@ function sendVideoUrlToMetubeAndSwitchTab(videoUrl, metubeUrl, tab) {
         if (res.ok === true && res.status === 200) {
             return res.json();
         }
-        // todo fix it
-        alert("error :: code " + res.status);
+        console.log("error :: code" + res.status);
     }).then(function(result) {
-        if (result.status === "ok") {
-            openTab(metubeUrl, tab);
-        } else {
-            // todo fix it
-            alert("error :: " + json);
+        if (result.status !== "ok") {
+            console.log("error :: ", result );
+            if (result.msg.includes('URLError')) {
+                // Go straight to catch block and skip the callback
+                throw new Error('Error when adding URL to MyTube');
+            }
         }
-    }).catch(function(res) {
-        alert("error :: " + res);
+    }).then(function() {
+        typeof callback === 'function' && callback();
```

Comment:
in another part of code we not use short style for expressions. if you don't mind, change that to `if` construction

Reply:
Sure